src/App.jsx

```diff
@@ -13,42 +13,64 @@ function App() {
   const [isLoading, setIsLoading] = useState(true)
   const [error, setError] = useState('')
 
-  useEffect(() => {
-    const myPromise = new Promise((resolve, reject) => {
-      setTimeout(() => {
-        resolve( JSON.parse(localStorage.getItem('savedTodoList')) || [] )
-      }, 2000)
-    })
-    myPromise.then((result) => {
-      setTodoList(result)
+  const fetchData = async () => {
+    const options = {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${import.meta.env.VITE_AIRTABLE_API_TOKEN}`
+      },
+      url: `https://api.airtable.com/v0/${import.meta.env.VITE_AIRTABLE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`,
+    }
+
+    try {
+      const response = await fetch(options.url, options)
+      if ( !response.ok ) {
+        throw new Error(`Error: ${response.status}`)
+      }
+      const data = await response.json()
+
+      const todos = data.records.map((item) => {
+        return {
+          id: item.id,
+          title: item.fields.title,
+          completedAt: item.fields.completedAt,
+        }
+      })
+      setTodoList(todos)
       setIsLoading(false)
-    })
-    
-    myPromise.catch((error) => {
+
```

Comment:
note: looks good. this doesn't directly apply here, but it's good practice to also setError to an empty string if the call is successful. If ever a fetch request happens on a button click rather than a page load